src/pyosmeta/file_io.py

```diff
@@ -21,124 +20,158 @@ def get_api_token() -> str:
     return os.environ["GITHUB_TOKEN"]
 
 
-@dataclass
-class YamlIO:
+def load_pickle(filename):
+    """Opens a pickle"""
+    with open(filename, "rb") as f:
+        return pickle.load(f)
+
+
+def _list_to_dict(a_list: list, a_key: str) -> dict:
+    """Takes a yaml file opened and turns into a dictionary
+    The dict structure is key (gh_username) and then a dictionary
+    containing all information for the username
+
+    a_list : list
+        A list of dictionary objects returned from load website yaml
+    a_key : str
+        A string representing the dict key to use as the main key to call
+        each sub dict in the object.
+
+    """
+    final_dict = {}
+    for dict in a_list:
```

Comment:
`dict` is a reserved word, so normally it is not a good practice to use that name.
also maybe you could use list comprehension here, something like:

`return { data[a_key].lower(): data for data in a_list }`

Reply:
that's much cleaner. i always forget about comprehensions for dictionaries and i'm not sure why.